src/main/java/com/minecolonies/coremod/entity/ai/citizen/beekeeper/EntityAIWorkBeekeeper.java

```diff
@@ -148,11 +151,21 @@ private IAIState prepareForHerding()
                 return getState();
             }
         }
+
         if (!getOwnBuilding().getHarvestTypes().equals(BuildingBeekeeper.HONEYCOMB))
         {
             checkIfRequestForItemExistOrCreateAsynch(new ItemStack(Items.GLASS_BOTTLE));
         }
-        checkIfRequestForTagExistOrCreateAsynch(ItemTags.FLOWERS, 16);
+
+        Set<Item> allowedFlowers = getOwnBuilding().getAllowedFlowers();
+        if (!InventoryUtils.hasItemInItemHandler(worker.getInventoryCitizen(), (stack) -> allowedFlowers.contains(stack.getItem()))
```

Comment:
if you keep the list of itemstorages you can search much easier for them,  as our inventories do contain itemstorages too

Reply:
how would I do that?
I tried looking for how that can be done, but I couldn't find it

Reply:
The predicate here should ratherbe "itemlistmodule.isItemInList", there is no need to collect the flowerlist here.

Then you can scrap the method in the building actually and down here just module.getList().mapStorageToStack

Reply:
dont we have some itemstorage specific functions which avoids us needing to make it stacks in the first place?

Reply:
seems we're missing some itemstorage predicate util for it

Reply:
Not for the citizen inventory, but there is a isInBuilding check afaik for the building.

Reply:
I will probably update this later today to `module.isItemInList` then.

Reply:
Like this?